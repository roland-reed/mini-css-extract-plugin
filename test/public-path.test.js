/* eslint-env browser */
import path from "path";

import MiniCssExtractPlugin from "../src/cjs";

import {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  runInJsDom,
} from "./helpers/index";

describe("public path", () => {
  it("should work with function output.publicPath", async () => {
    const compiler = getCompiler(
      "simple.js",
      {},
      {
        output: {
          publicPath() {
            return "";
          },
          path: path.resolve(__dirname, "../outputs"),
          filename: "[name].bundle.js",
        },
        plugins: [
          new MiniCssExtractPlugin({
            filename: "[name].css",
          }),
        ],
      }
    );
    const stats = await compile(compiler);

    runInJsDom("main.bundle.js", compiler, stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot("DOM");
    });

    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
