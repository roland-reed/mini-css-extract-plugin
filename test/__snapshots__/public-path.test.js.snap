// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`public path should work with function output.publicPath: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
</head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`public path should work with function output.publicPath: errors 1`] = `Array []`;

exports[`public path should work with function output.publicPath: warnings 1`] = `Array []`;
